### PR TITLE
[invite] Add implementation for invitation record methods

### PR DESCRIPTION
**Pull request by rcebulko**
```
 PASS  test/invitation_record.test.ts
  invitation record
    recordInvite
      ✓ records an invite (5ms)
      ✓ sets `archived = false` (1ms)
    getInvites
      if no invite record exists for the user
        ✓ returns an empty list (2ms)
      if records exists of invites to the user
        ✓ returns the invites (1ms)
      if only archived invite records exists for the user
        ✓ returns an empty list (1ms)
    archiveInvites
      if records exists of any invites to the user
        ✓ updates the invite records (3ms)
```

### CODE REVIEW

File: invite/src/types.ts [hunk 1]
```diff
@@ -40,5 +40,5 @@ export interface Invite {
   repo: string;
   issue_number: number;
   action: InviteAction;
-  archived?: boolean;
+  archived?: boolean | number; // stored as a TINYINT in PostgresQL
```
Comment (estherkim) [inline invite/src/types.ts]: This is a little scary! The interface should stay as a boolean but be converted when retrieving or storing SQL data.

Comment (rcebulko) [reply invite/src/types.ts]: I've dropped the `number` type and added explicit casting in the method that actually fetches from the DB.

Thanks for the link! Would it be alright with you if I move this to a declaration file in a separate PR? There are a handful of files that import it directly, and `Database` type could be moved into the type declaration file as well. I'd prefer to do that as a separate set of changes to minimize rebase headaches

Comment (rcebulko) [reply invite/src/types.ts]: https://github.com/ampproject/amp-github-apps/pull/703

Comment (estherkim) [reply invite/src/types.ts]: Sounds good!

Comment (rcebulko) [reply invite/src/types.ts]: Given that, is this PR LGTY?

**Pull request merged**